guide/english/javascript/destructuring/index.md

```diff
@@ -0,0 +1,108 @@
+---
+title: Destructuring
+---
+## Destructuring Assignment
+The destructuring assignment syntax is a way to "unpack" values from arrays, properties from objects, or any other iterable, into distinct variables.
+Since ```Arrays``` and ```Objects``` are the most commonly data structures used in JavaScript then the usage of destructuring can be extremely handy for data manipulation and code clarity.
+
+## Array Destructuring
+Given the array below
+``` javascript
+const fruits = ['apple', 'banana', 'orange', 'tangerine'];
+```
+The tranditional way of getting these values into separate variables would be:
+``` javascript
+const firstFruit = fruits[0];   // apple
+const secondFruit = fruits[1];  // banana
+const thirdFruit = fruits[2];   // orange
+const fourthFruit = fruits[3];  // tangerine
+```
+With destructuring this can be a one-liner assignment:
+``` javascript
+const [firstFruit, secondFruit, thirdFruit, fourthFruit] = fruits;
+```
+This assigns in every variable from the left-side the corresponding value from the array on the right in order of appearance, meaning we get the same result as the tranditional way.
+
+We can also get the first element of the array when we don't care about the rest:
+``` javascript
+const [firstFruit] = fruits;
+```
+Or just the second by ignoring the first one and so on:
+``` javascript
+const [, secondFruit] = fruits;  //ignores the first fruit
+const [,, thirdFruit] = fruits;  //ignores first and second fruit
+```
+
+Additionaly we can use the <a href='https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters' target='_blank' rel='nofollow'>`rest-operator`</a> to get some values and then pack the remaining in another variable:
+``` javascript
```

Comment:
Can you change `rest-operator` to `rest-parameter`? In [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters#Destructuring_rest_parameters) and ES spec, its mentioned as rest parameter so we should stick to that.

Reply:
done